src/main/java/org/elasticsearch/common/geo/builders/BasePolygonBuilder.java

```diff
@@ -112,6 +116,18 @@ public ShapeBuilder close() {
     }
 
     /**
+     * Validates only 1 vertex is tangential (shared) between the interior and exterior of a polygon
+     */
+    protected void validateHole(BaseLineStringBuilder shell, BaseLineStringBuilder hole) {
+        ArrayList exterior = Lists.newArrayList(Sets.newHashSet(shell.points));
```

Comment:
I noticed HashSet inherits retainAll from AbstractCollection. So I eliminated the intermediate ArrayList and used it directly to avoid both unnecessary memory use and having to sort.